Sources/NIOHPACK/HPACKHeader.swift

```diff
@@ -19,10 +19,28 @@ import NIOHTTP1
 /// information.
 public struct HPACKHeaders {
     internal var headers: [HPACKHeader]
-    
+
+    // see 8.1.2.2. Connection-Specific Header Fields in RFC 7540
+    let illegalHeaders: [String] = ["connection", "keep-alive", "proxy-connection", "transfer-encoding", "upgrade"]
+
+    /// Constructor that can be used to map between `HTTPHeaders` and `HPACKHeaders` types.
+    public init(httpHeaders: HTTPHeaders, normalizeHTTPHeaders: Bool) {
+        self.headers = httpHeaders.map { HPACKHeader(name: normalizeHTTPHeaders ? $0.name.lowercased() : $0.name,
+                                                     value: $0.value) }
```

Comment:
Rather than branch on `normalizeHTTPHeaders` every time it may be better to have a single `if` that sets up the header block differently.

Reply:
Oh hey, we already have that `if` block below, so let's do that. That also allows us to rewrite this code to make only a single pass over the headers map, instead of doing two.

Reply:
done, thanks